apps/web/views.py

```diff
@@ -35,21 +35,30 @@ def internal_server_error(request):
 
 
 @throttle_classes([AnonRateThrottle, ])
-@api_view(['POST', ])
+@api_view(['POST', 'GET'])
 @permission_classes((permissions.AllowAny,))
 def contact_us(request):
     try:
-        user = User.objects.get(username=request.user)
-        name = user.username
-        email = user.email
-        request_data = {"name": name, "email": email}
-        request_data['message'] = request.data['message']
-        serializer = ContactSerializer(data=request_data)
-        if serializer.is_valid():
-            serializer.save()
-            response_data = {'message': 'Your message has been successfully recorded. We will contact you shortly.'}
-            return Response(response_data, status=status.HTTP_201_CREATED)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        if request.method == 'POST':
+            user = User.objects.get(username=request.user)
+            name = user.username
+            email = user.email
+            request_data = {"name": name, "email": email}
+            request_data['message'] = request.data['message']
+            serializer = ContactSerializer(data=request_data)
+            if serializer.is_valid():
+                serializer.save()
+                response_data = {'message': 'Your message has been successfully recorded. We will contact you shortly.'}
+                return Response(response_data, status=status.HTTP_201_CREATED)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        elif request.method == 'GET':
+            user = User.objects.get(username=request.user)
```

Comment:
We would probably remove the `try` statement from line 41 and make it more specific to the particular statement. also, if try-except block is required for some other piece of code, then you should add a separate try-catch for that statement.